### Step 1: Import Libraries & API Keys

In [ ]:
import os
from openai import OpenAI
from dotenv import load_dotenv
from IPython.display import Markdown, display

load_dotenv()

OPENAI_API_KEY = os.getenv("OPENAI_API_KEY")

if OPENAI_API_KEY is None:
    raise Exception("API key is missing.")

client = OpenAI(api_key=OPENAI_API_KEY)

### Step 2: Searching the web

In [5]:
completion = client.chat.completions.create(
    model="gpt-4o-mini-search-preview",
    messages=[
        {"role": "user", "content": "What is today's date? Tell me the 3 top stories in AI in this past week, according to reliable news media, do not use simple news aggregators. Each story should be from a separate source and date. Include dates."}
    ],
    max_tokens=500
)


In [6]:
search_results = completion.choices[0].message.content
display(Markdown(search_results))

Today's date is Monday, May 18, 2026. Here are three significant AI developments from the past week, each reported by a different reputable source:

1. **Anthropic Surpasses OpenAI in Enterprise AI Adoption**

   On May 13, 2026, Ramp's AI Index reported that Anthropic's AI system, Claude Code, has achieved 34.4% enterprise adoption, surpassing OpenAI's 32.3%. This shift indicates a growing preference for Anthropic's AI solutions in business applications. ([champaignmagazine.com](https://champaignmagazine.com/2026/05/17/ai-by-ai-weekly-top-5-may-11-17-2026/?utm_source=openai))

2. **Yale Introduces Governance Framework for Agentic AI Systems**

   On May 11, 2026, Yale's Chief Executive Leadership Institute unveiled a new governance framework addressing the risks and controls associated with autonomous AI systems. The framework emphasizes accountability, decision reversibility, transparency, and stakeholder impact, reflecting increasing concerns about managing AI agents capable of independent actions. ([aileaderscouncil.org](https://aileaderscouncil.org/ai-insiders-weekly-top-stories-may-11th-edition/?utm_source=openai))

3. **FDA Approves 1,451 AI Devices for Medical Use**

   On May 13, 2026, the U.S. Food and Drug Administration (FDA) announced the approval of 1,451 AI-powered medical devices. This milestone underscores the growing integration of AI technologies in healthcare, enhancing diagnostic accuracy and patient care. ([justoborn.com](https://justoborn.com/ai-weekly-news-88/?utm_source=openai))


## Highlights:
- [Android Show 2026 as it happened - all the big news on Android 17, Googlebook, and Gemini Intelligence](https://www.techradar.com/news/live/android-show-2026-live?utm_source=openai), Published on Tuesday, May 12
- [Top tech conferences: The ultimate tech events guide for May 2026](https://www.techradar.com/news/top-tech-conferences-the-ultimate-tech-events-and-show-guide?utm_source=openai), Published on Monday, May 11
- [Google I/O 2026: Date, time, potential announcements and everything else you need to know](https://www.tomsguide.com/phones/google-pixel-phones/google-i-o-2026-date-time-potential-announcements-and-everything-else-you-need-to-know?utm_source=openai), Published on Monday, May 11 

### Step 3: Summarize and write social media post

In [7]:
system_message = """
You are a helpful assistant who creates amazing LinkedIn posts about AI based on text that you are provided with. Make the post engaging and interesting.
Also make it concise and add a thought-provoking question at the end to encourage comments.
"""

prompt = f"""
The following text containts top 3 news stories about AI from the past week.
Now please write my LinkedIn post based on this information. Make it engaging and interesting.
{search_results}
"""

completion = client.chat.completions.create(
    model="gpt-4o-mini",
    messages=[
        {"role": "system", "content": system_message},
        {"role": "user", "content": prompt}
    ],
    max_tokens=500,
    temperature=0.7
)

chat_response = completion.choices[0].message.content

In [8]:
print(chat_response)

🚀 Exciting times in the world of AI! Here are three major developments from the past week that are shaping the future of this transformative technology:

1. **Anthropic Emerges as a Leader in Enterprise AI**  
   Anthropic's AI system, Claude Code, has surpassed OpenAI in enterprise adoption, achieving a notable 34.4%. This shift signals a growing preference for Anthropic's solutions in business applications, highlighting an evolving landscape in enterprise AI. [Read more](https://champaignmagazine.com/2026/05/17/ai-by-ai-weekly-top-5-may-11-17-2026/?utm_source=openai)

2. **Yale Sets the Standard for Governance in AI**  
   Yale's Chief Executive Leadership Institute has introduced a groundbreaking governance framework for agentic AI systems. This framework addresses critical issues like accountability and transparency, reflecting the urgent need for responsible AI management as it becomes more autonomous. [Learn more](https://aileaderscouncil.org/ai-insiders-weekly-top-stories-may-11

🚀 Exciting times in the world of AI! Here are three major developments from the past week that are shaping the future of this transformative technology:

1. **Anthropic Emerges as a Leader in Enterprise AI**  
   Anthropic's AI system, Claude Code, has surpassed OpenAI in enterprise adoption, achieving a notable 34.4%. This shift signals a growing preference for Anthropic's solutions in business applications, highlighting an evolving landscape in enterprise AI. [Read more](https://champaignmagazine.com/2026/05/17/ai-by-ai-weekly-top-5-may-11-17-2026/?utm_source=openai)

2. **Yale Sets the Standard for Governance in AI**  
   Yale's Chief Executive Leadership Institute has introduced a groundbreaking governance framework for agentic AI systems. This framework addresses critical issues like accountability and transparency, reflecting the urgent need for responsible AI management as it becomes more autonomous. [Learn more](https://aileaderscouncil.org/ai-insiders-weekly-top-stories-may-11th-edition/?utm_source=openai)

3. **FDA Approves a Surge of AI Medical Devices**  
   In a remarkable advancement for healthcare, the FDA has approved 1,451 AI-powered medical devices, enhancing diagnostic accuracy and patient care. This milestone underscores the significant role AI is playing in revolutionizing the medical field. [Discover more](https://justoborn.com/ai-weekly-news-88/?utm_source=openai)

As AI continues to integrate into various sectors, the implications for our future are immense. How do you think we can ensure that AI remains beneficial and ethical as it evolves? 🤖💡 #AI #Innovation #Healthcare #Governance #EnterpriseAI